shm/src/Time.cpp

```diff
@@ -0,0 +1,30 @@
+#include "shm/inc/Time.hpp"
+
+#include <algorithm>
+
+#include "shm/inc/Subscriber.hpp"
+
+void Time::addSubscriber(Subscriber* subscriber) {
+    subscribers_.push_back(subscriber);
```

Comment:
What if a subcriber already exists? We will add him twice and had a lot of problems with debugging. Validate if you already add some pointers here